### PR TITLE
Reproduce Bug 10523

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/10523
closes https://github.com/phpstan/phpstan/issues/10822

### CODE REVIEW

File: src/Node/ClassPropertiesNode.php [hunk 1]
```diff
@@ -355,6 +359,12 @@ private function getMethodsCalledFromConstructor(
 				continue;
 			}
 
+			if ($inMethod->getName() !== '__construct') {
```
Comment (staabm) [inline src/Node/ClassPropertiesNode.php:360]: at this point we take the properties which have been assigned in `__construct` over to the `additionalConstructor` methods as described in https://github.com/phpstan/phpstan/issues/10523#issuecomment-1923616806.

**Comment by staabm** [comment]
I think this one is ready for a review

**Comment by ondrejmirtes** [comment]
This also fixes https://github.com/phpstan/phpstan/issues/10822 please add a regression test for that.

**Comment by staabm** [comment]
@ondrejmirtes here you are

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**